scipy/linalg/basic.py

```diff
@@ -360,12 +360,38 @@ def solve_triangular(a, b, trans=0, lower=False, unit_diagonal=False,
                      (-info))
 
 
+def _diagonal_banded(l_and_u, a):
+    n = a.shape[1]
+    if a.shape != (n, n):
+        raise ValueError("Matrix must be square (has shape %s)" % (a.shape,))
+    (nlower, nupper) = l_and_u
+    if nlower > n or nupper > n:
+        raise ValueError("Number of nonzero diagonals must be less than square dimension")
```

Comment:
Neither this exception nor the one above are covered by unit tests are the moment.

Reply:
Unit tests updated :)